consumer/simple/metrics_test.go

```diff
@@ -0,0 +1,432 @@
+// Copyright 2020 The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package simple
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/testutil/metricstestutil"
+)
+
+func ExampleMetrics() {
+	metrics := pdata.NewMetrics()
+
+	mb := Metrics{
+		Metrics:                       metrics,
+		InstrumentationLibraryName:    "example",
+		InstrumentationLibraryVersion: "0.1",
+		ResourceAttributes: map[string]string{
+			"host": "my-host",
+		},
+		Timestamp: time.Now(),
+	}
+
+	for _, disk := range []string{"sda", "sdb", "sdc"} {
+		// All metrics added after this will have these labels
+		diskBuilder := mb.WithLabels(map[string]string{
+			"disk": disk,
+		}).AddGaugeDataPoint("disk.usage", 9000000)
+
+		// Add metrics in a chained manner
+		diskBuilder.
+			AddGaugeDataPoint("disk.capacity", 9000000).
+			AddDGaugeDataPoint("disk.temp", 30.5)
+
+		// Add additional labels
+		diskBuilder.WithLabels(map[string]string{
+			"direction": "read",
+		}).AddSumDataPoint("disk.ops", 50)
+
+		diskBuilder.WithLabels(map[string]string{
+			"direction": "write",
+		}).AddSumDataPoint("disk.ops", 80)
+	}
+
+	metricCount, dpCount := metrics.MetricAndDataPointCount()
+	fmt.Printf("Metrics: %d\nDataPoints: %d", metricCount, dpCount)
+
```

Comment:
Would you be able to show the expected usage w/ the next consumer's `ConsumeMetrics()` and how this could work over an interval?  Are we to be creating a scratch `Metrics{}` with every pass or would an exported `Clone()` help?

Reply:
ConsumeMetrics just takes `metrics` as an argument.

Yes you would need to make a new builder and pdata.Metrics instance each time as sharing metric data between calls to ConsumeMetrics is forbidden.

Reply:
Thanks, mind adding equivalent remarks to this example for clarity?